plugins/woocommerce/client/blocks/assets/js/atomic/blocks/product-elements/button/frontend.ts

```diff
@@ -51,14 +55,58 @@ const { state: addToCartWithOptionsState } = store< AddToCartWithOptionsStore >(
 	{ lock: universalLock }
 );
 
+const isCartItemMatched = (
+	cartItem: OptimisticCartItem,
+	selectedItem: CartVariationItem[]
+) => {
+	if (
+		! Array.isArray( cartItem.variation ) ||
+		! Array.isArray( selectedItem )
+	)
+		return false;
```

Comment:
Small formatting suggestion, but we usually try to wrap clauses with `{ ... }`:

```suggestion
	if (
		! Array.isArray( cartItem.variation ) ||
		! Array.isArray( selectedItem )
	) {
		return false;
	}
```